### DISCUSSION TITLE
Are there any resources how PMTiles relates to other formats?

### DISCUSSION THREAD

**Discussion by wiesehahn** [protomaps/PMTiles#85] [Q&A] [answered]
I am sure there is information about this somewhere out there but I am always getting confused.
Is it possible or does it make sense to use COGs as underlaying raster tiles in PMTiles? Whats the underlaying vector tile format? Is is MBTiles and how does this relate to non-tiled formats like flatgeobuf or geoparquet? How does PMTiles relate to COM-Tiles? Which are good use cases for MBTiles and which are not? ...

What are good starting points to get information on these questions?

**Comment by bdon** [comment, marked as answer]
Feels like we need a "PMTiles vs COG" doc somewhere ... :)

PMTiles is both more specific and more general than COG:

* COG is specific to raster images in TIFF format, PMTiles is for arbitrary tiled data.
* COG is a subset of Tiff with internally tiled organization
* PMTiles is specific to one particular tiling scheme - pyramidal Z/X/Y - that is often mapped to the "Web Mercator" cartographic projection. COG instead specifies metadata about its spatial reference system, which makes it more powerful but more complex
* PMTiles scales to much larger (10s-100s of GB) datasets than COG because of partitioned directories
* PMTiles only specifies the addressing of tiles itself, not the format of individual tiles. Those could be small square TIFFs, JPEGs, PNGs. The most popular vector format is the Protobuf format [Mapbox Vector Tile](https://github.com/mapbox/vector-tile-js) but now supported by other systems like PostgreSQL unrelated to Mapbox

On alternatives:
* MBTiles is a 1:1 alternative to PMTiles. MBTiles cannot be accessed on remote static cloud storage the way PMTiles offers, but MBTiles has more existing tooling, is interoperable with any SQLite client, and is a transactional database
* [COM-Tiles](https://github.com/mactrem/com-tiles) and [TileBase](https://github.com/openaddresses/TileBase) are also essentially 1:1 alternatives to PMTiles, though have different internal index designs that may lead to different tradeoffs in performance or storage space. 

This discussion area on GitHub is the best place for info while I roll out more documentation.

**Reply by wiesehahn** [reply]
Thanks, this clarifies some aspects. Nevertheless I would definitely appreciate something like  "PMTiles vs COG" or "PMTiles hands-on" :)
I guess updating PMTiles (e.g. changing a single tile) is not possible without rewriting the entire dataset?

**Reply by bdon** [reply]
@wiesehahn correct, you need to re-write the entire thing, though the structure should make re-writing fairly easy.

**Reply by wiesehahn** [reply]
you state that "PMTiles scales to much larger (10s-100s of GB) datasets". I use COGs which are 30-50 GB, so theoretical its possible. 
What are the benefits of PMTiles for raster data? Is the query more efficient and thus faster?

**Reply by bdon** [reply]
@wiesehahn I work mostly with vector data so that is good to know! In theory it should be more efficient if you are visualizing on a ZXY web map, unless your COG's internal tiling scheme and overview exactly match Z/X/Y tiles. I'm not familiar enough with COG to know if this is possible or easy to guarantee. 

This might have more specifics: https://developmentseed.org/blog/2019-05-23-cog-talk-part-2-mosaics

**Reply by wiesehahn** [reply]
thanks, 
one downside why i am struggling to use PMTiles is that it is only possible to encode MBTiles through gdal as jpeg or png. But most efficient option to store elevation data I found is an rgb-encoded elevation data set compressed as WEBP. E.g. in my case the (WebP) COG is 27GB, the (PNG) MBTiles are 83GB and the PMTiles are 79GB (I have to admit I did not try to optimize this as the conversion to MBTiles takes veryyy long)

**Reply by bdon** [reply]
Do you think the difference is due to image encoding (WebP vs PNG) or is it due to the different tiling between COG and MB/PMTiles?

**Reply by wiesehahn** [reply]
I think its just the encoding (WebP vs PNG) and maybe the data type (byte in case of COG).

**Reply by bdon** [reply]
I've updated the FAQ at https://protomaps.com/docs/faq#cloud-optimized-geotiffs with more information on how PMTiles relates to other formats.

**Comment by farfromrefug** [comment]
@bdon would it be possible to have a little more numbers on pmtiles vs mbtiles. Especially size comparaison for medium/large mbtiles. Is it similar, bigger or smaller ? 
Also do you have any idea if requesting tiles, zooming in out would be slower with pmtiles  ? I am investigating pmtiles for offline usage on android/iPhone devices. One gain I would already see is that sqlite library is quite big so it would be a gain. What I don't really envision right now is how easy/ fast it is to read part of a file at an offset.
Thanks

**Reply by bdon** [reply]
It depends on your data and application. PMTiles should always be smaller, but only slightly for general datasets; the greatest size reduction will be for deep zooms (z13+) with large polygons. 

It's impossible to generalize about performance for the file formats, since there are usually implementation and application-specific caches in use. Your android/iPhone app should be able to use SQLite as part of the OS library. If you're just reading from disk and not using a file over HTTP then I don't see a great reason to use PMTiles over MBTiles.

**Reply by farfromrefug** [reply]
@bdon thanks a lot for the explanations. I might at one point try to convert a mbtiles(zoom 0-14) to compare size!
Thanks again
